sdk/core/Azure.Core/src/IJsonSerializable.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Azure
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public interface IJsonSerializable
+    {
+        /// <summary>
+        /// TODO
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="bytesWritten"></param>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        bool TrySerialize(Stream stream, out long bytesWritten, SerializableOptions? options = default);
```

Comment:
I appreciate we don't want to use exceptions for code flow, but is there any *expected* scenario where we wouldn't serialize when we're told (via call) that we should serialize? If not, this just creates undue burden and an exception is warranted.

Reply:
Is the suggestion here to have a `Serialize` and `Deserialize` that throw an exception along side the Try variants?

Reply:
If an error is truly exceptional, yes. If you're just returning a `false` when an error occurs and expect the caller to throw their own error - leading to inconsistencies - just throw. Are there any error cases that would be expected e.g., completely empty body - no `{}` or `[]`, just empty?

Reply:
as long as its valid json there shouldn't be an error

Reply:
Do we have an example usage scenario that would help us understand the value of using TryXx() vs. Non-try APIs?

Reply:
+1 for the "no try, just do"

As Anne mentioned, I'm having trouble figuring out a scenario where I would try.   Are you thinking of something like "I don't know what type this is, so I'll try A and if it doesn't work, then B, and if that doesn't work, it must be C?"

Reply:
> As Anne mentioned, I'm having trouble figuring out a scenario where I would try. 

Ah, for clarification, I was asking my question quite honestly - I believe such a scenario may exist, I would just like to understand it more concretely, and I think it would be useful for all of us to look at for the purposes of evaluation.  I was not trying to imply with my question that such a scenario does not exist.

In general, TryXx() APIs tend to be provided to make it possible to call APIs without invoking the exception handling machinery, which is bad for performance.  If Try APIs exist, it's easy to implement non-Try APIs around them, but not vice versa.  So in principle I have no problem with using Try APIs (I tend to like them), but it does feel like a non-standard choice for an interface.  And also, as @JoshLove-msft mentioned elsewhere, they don't work with async methods because of out parameters, so if that's a need here, we may not have that option.

Reply:
Apologies if it came across otherwise; it was not my intent to imply that you didn't think there were scenarios.   My assumption was that you were also having difficulty coming up with a real-world example.

Reply:
The example usage is in the test case in this PR.  For easy reference https://github.com/Azure/azure-sdk-for-net/pull/35742/files#diff-46013d0356c5137faeddc1f0b2c084532794fcea22ff8574fa4a46b824640855R46-R47

Reply:
I see it's used, but I still don't understand why. If there is any exception writing object data to a stream of bytes, it seems exceptional and should throw an exception. Is there a case where any underlying exception - or even a higher-level condition - should simply return false? And if there is a case where the underlying exception is thrown - like failure to write to a stream - shouldn't the caller know the details instead of having to get back `false` and then throw some exception themselves in a myriad of ways sans details?